### ISSUE TITLE
Option to insert CSV from command line

### ISSUE THREAD

**Issue by jevin36** [jgraph/drawio-desktop#202]
I would very much like a way how I can programatically generate diagrams (e.g. in XML) from with my programming language of choice.

I had a look at the exported XML but that seems to be very complicated to generate that. Later I saw the GUI feature in Arrange-->Insert-->Advanced-->CSV that supports importing defined diagrams from csv but unfortunately the option is missing in the command line client

**Comment by CrashLaker** [comment]
also looking for that

**Comment by tpelado** [comment]
Hello,

I'm writing on this cause I'm facing the same "issue" of sort. Being able to automate the creation of a drawing from a csv file with the CLI tools would be very useful.  


Cheers

**Comment by davidjgraph** [comment]
Please thumbs up the original post rather than writing "me too"

**Comment by zivklara** [comment]
that would be very useful!

**Comment by sailhobie** [comment]
bump! this would be a great option

**Comment by OlegLoewen** [comment]
I would really appreciate this feature!

**Comment by OlegLoewen** [comment]
if you tell me where to begin I can provide a pull request.

**Comment by empusas** [comment]
#202 and #8ß7 referring to each other and both are closed without further comment. So what about that request?

**Comment by davidjgraph** [comment]
This is 202 and it's not closed. I don't understand your point.

**Comment by neeles83** [comment]
This would be would be a great option indeed.

**Comment by mostsignificant** [comment]
I would appreciate this too - it would be an important feature for generating diagrams in a CI/CD pipeline

**Comment by m-mohamedin** [comment]
Will be available in the next release
Usage (the same but input file extension is .csv):

-x file.csv -f [xml/pdf/png/jpg/svg]

**Comment by MichaelCropper** [comment]
@m-mohamedin Excellent news. Do you have a link to the documentation for this new feature so I can have a play once it is released?

**Comment by m-mohamedin** [comment]
> @m-mohamedin Excellent news. Do you have a link to the documentation for this new feature so I can have a play once it is released?

Nothing special, just the input file is *.csv file instead of *.drawio

**Comment by davidjgraph** [comment]
Internal commit https://github.com/jgraph/drawio-dev/commit/fd04104ee1a48594a5248bcbf637f67139689a23

**Comment by fi5sh** [comment]
> > @m-mohamedin Excellent news. Do you have a link to the documentation for this new feature so I can have a play once it is released?
> 
> Nothing special, just the input file is *.csv file instead of *.drawio

@m-mohamedin I've tried to use `drawio input.csv -c` but I get invalid file errors. How to do this on the command line?
I've used the csv files from https://drawio-app.com/import-from-csv-to-drawio/

**Comment by m-mohamedin** [comment]
> > > @m-mohamedin Excellent news. Do you have a link to the documentation for this new feature so I can have a play once it is released?
> > 
> > 
> > Nothing special, just the input file is *.csv file instead of *.drawio
> 
> @m-mohamedin I've tried to use `drawio input.csv -c` but I get invalid file errors. How to do this on the command line? I've used the csv files from https://drawio-app.com/import-from-csv-to-drawio/

Sorry but the new version not released yet

**Comment by WilliamStenberg** [comment]
`drawio -x input.csv` works, generates a PDF file (default option for `-x`).
`drawio input.csv`, still invalid file errors, expected to open DrawIO with the generated content.

Workaround for my use case is to generate the file but specify XML format, and then open the XML file:
`drawio -x input.csv -f xml`
`drawio input.xml`

Tried after installing the [v20.7.4](https://github.com/jgraph/drawio-desktop/releases/tag/v20.7.4) RPM.

**Comment by glisterGIT** [comment]
Hi i am desperate to get the CSV import to work from the command line I am using the latest desktop version 21.6.5 but if I try something like this below I get no errors and no output:
 & "C:/Program Files/draw.io/draw.io.exe" -x C:\Temp\draw\test.csv -f xml
:( is there a step by step guide ??

**Comment by emptaj** [comment]
> Hi i am desperate to get the CSV import to work from the command line I am using the latest desktop version 21.6.5 but if I try something like this below I get no errors and no output:
& "C:/Program Files/draw.io/draw.io.exe" -x C:\Temp\draw\test.csv -f xml
:( is there a step by step guide ??

I get the same result. With 20.7.4 everything works fine, but with 21.6.5 there is no errors/outputs - process stuck in background

**Comment by glisterGIT** [comment]
could it be a bug do you think
im not sure where to log it...

**Comment by emptaj** [comment]
I've tested different versions: 21.6.5, 21.6.1, 21.5.1 and 20.7.4. It works only for the last one. We can create another ticket in https://github.com/jgraph/drawio-desktop/issues section - this place is dedicated for reporting issues

**Comment by glisterGIT** [comment]
yep tested myself and it works on 20.7.4, but I can only create xml or pdf.  I am really after html
